frontend/__tests__/api/books.test.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { useQuery, useMutation } from "@apollo/client";
+import { render, screen } from "../../test-utils";
+import {
+  useGetBooks,
+  useGetBook,
+  useAddBook,
+  useRemoveBook,
+} from "../../api/books";
+
+jest.mock("@apollo/client");
+
+describe("Books test", () => {
+  const BooksTest = () => {
+    const { books } = useGetBooks();
+    const { book } = useGetBook();
+    const { addBook } = useAddBook();
+    const { removeBook } = useRemoveBook();
+    addBook();
+    removeBook();
+
+    return books && book ? <div>test</div> : null;
+  };
+  it("should use books hook without error", () => {
+    const data = {};
+    useQuery.mockReturnValue({ loading: false, error: null, data });
+    useMutation.mockReturnValue([
+      () => {},
+      { loading: false, error: null, data },
+    ]);
+    render(<BooksTest />);
+
+    expect(screen.getByText("test")).toBeInTheDocument();
+  });
+});
```

Comment:
Please re-examine your testing approach when testing things that don't necessarily have UI to render. Also please take a look at how you are using mocking. (line 6)